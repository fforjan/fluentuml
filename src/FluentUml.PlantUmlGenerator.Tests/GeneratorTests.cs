
using Microsoft.VisualStudio.TestTools.UnitTesting;
using FluentUml.SelfDescribe;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;

[assembly: UseReporter(typeof(DiffReporter))]

namespace FluentUml.PlantUmlGenerator.Tests;
[TestClass]
public class GeneratorTests
{
    [TestMethod]
    public void ComponentDiagramTest()
    {
        // Arrange
        var generator = new Generator();
        var diagram = new ComponentDiagram();
        diagram.Describe();

        using var textStream = new StringWriter();
        
        // Act
        generator.GeneratePlantUmlDocument(diagram, textStream);

        // Assert
        Approvals.Verify(textStream.GetStringBuilder().ToString());
    }

    [TestMethod]
    public void MEFDiagramTest()
    {
        // Arrange
        var generator = new Generator();
        var diagram = new MEFDiagram();
        diagram.Describe();
        
        using var textStream = new StringWriter();
        
        // Act
        generator.GeneratePlantUmlDocument(diagram, textStream);

        // Assert
        Approvals.Verify(textStream.GetStringBuilder().ToString());
    }
}