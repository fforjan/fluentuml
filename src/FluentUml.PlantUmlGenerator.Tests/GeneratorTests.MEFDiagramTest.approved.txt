@startuml
package FluentUml.SelfDescribe {

DiagramBase -- [ComponentDiagram]

DiagramBase -- [MEFDiagram]
}
package FluentUml.Generator {
}
package FluentUml.Model {
}
package FluentUml.ExcelGenerator {
}
package FluentUml.PlantUmlGenerator {

IGenerator -- [Generator]
}
@enduml
